auth0/resource_auth0_client.go

```diff
@@ -753,130 +750,157 @@ func expandClient(d *schema.ResourceData) *management.Client {
+		for key, value := range clientMetadata.(map[string]interface{}) {
+			client.ClientMetadata[key] = value.(string)
 		}
 	}
 
 	List(d, "native_social_login").Elem(func(d ResourceData) {
-		c.NativeSocialLogin = &management.ClientNativeSocialLogin{}
+		client.NativeSocialLogin = &management.ClientNativeSocialLogin{}
 
 		List(d, "apple").Elem(func(d ResourceData) {
 			m := make(MapData)
-			m.Set("enabled", Bool(d, "enabled"))
+			result = multierror.Append(result, m.Set("enabled", Bool(d, "enabled")))
 
-			c.NativeSocialLogin.Apple = m
+			client.NativeSocialLogin.Apple = m
 		})
 
 		List(d, "facebook").Elem(func(d ResourceData) {
 			m := make(MapData)
-			m.Set("enabled", Bool(d, "enabled"))
+			result = multierror.Append(result, m.Set("enabled", Bool(d, "enabled")))
 
-			c.NativeSocialLogin.Facebook = m
+			client.NativeSocialLogin.Facebook = m
 		})
 	})
 
 	List(d, "mobile").Elem(func(d ResourceData) {
-
-		c.Mobile = make(map[string]interface{})
+		client.Mobile = make(map[string]interface{})
 
 		List(d, "android").Elem(func(d ResourceData) {
 			m := make(MapData)
-			m.Set("app_package_name", String(d, "app_package_name"))
-			m.Set("sha256_cert_fingerprints", Slice(d, "sha256_cert_fingerprints"))
+			result = multierror.Append(
+				result,
+				m.Set("app_package_name", String(d, "app_package_name")),
+				m.Set("sha256_cert_fingerprints", Slice(d, "sha256_cert_fingerprints")),
+			)
 
-			c.Mobile["android"] = m
+			client.Mobile["android"] = m
 		})
 
 		List(d, "ios").Elem(func(d ResourceData) {
 			m := make(MapData)
-			m.Set("team_id", String(d, "team_id"))
-			m.Set("app_bundle_identifier", String(d, "app_bundle_identifier"))
 
-			c.Mobile["ios"] = m
+			result = multierror.Append(
+				result,
+				m.Set("team_id", String(d, "team_id")),
+				m.Set("app_bundle_identifier", String(d, "app_bundle_identifier")),
+			)
+
+			client.Mobile["ios"] = m
 		})
 	})
 
-	return c
+	return client, result.ErrorOrNil()
 }
 
-func buildClientAddon(d map[string]interface{}) map[string]interface{} {
-
-	addon := make(map[string]interface{})
-
-	for key, value := range d {
+func mapFromState(input map[string]interface{}) map[string]interface{} {
+	output := make(map[string]interface{})
 
+	for key, value := range input {
 		switch v := value.(type) {
-
 		case string:
 			if i, err := strconv.ParseInt(v, 10, 64); err == nil {
-				addon[key] = i
+				output[key] = i
 			} else if f, err := strconv.ParseFloat(v, 64); err == nil {
-				addon[key] = f
+				output[key] = f
 			} else if b, err := strconv.ParseBool(v); err == nil {
-				addon[key] = b
+				output[key] = b
 			} else {
-				addon[key] = v
+				output[key] = v
 			}
-
 		case map[string]interface{}:
-			addon[key] = buildClientAddon(v)
-
+			output[key] = mapFromState(v)
 		case []interface{}:
-			addon[key] = v
+			output[key] = v
+		default:
+			output[key] = v
+		}
+	}
+
+	return output
+}
 
+func mapToState(input map[string]interface{}) map[string]interface{} {
```

Comment:
In order to prevent us from introducing a breaking change we need to ensure that the `TypeMap` attribute always contains strings when in the state but actually use correct types when sending it to the Auth0 management API. We added a `mapToState` that will convert types like integers, bools, float to strings so all `TypeMap` key values are strings.